blockmanager.go

```diff
@@ -206,6 +217,18 @@ type blockManager struct { // nolint:maligned
 	minRetargetTimespan int64 // target timespan / adjustment factor
 	maxRetargetTimespan int64 // target timespan * adjustment factor
 	blocksPerRetarget   int32 // target timespan / target time per block
+
+	// hdrTipToResponse is a map that holds the response gotten from querying peers
+	// using the workmanager, to fetch headers within the chain's checkpointed region.
+	// It is a map of the request's startheight to the fetch response.
+	hdrTipToResponse map[int32]*headersMsg
+
+	// hdrTipSlice is a slice that holds request startHeight of the responses that have been
+	// fetched using the workmanager to fetch headers within the chain's checkpointed region.
+	// It is used to easily access this startheight in the case we have to delete these responses
+	// in the hdrTipResponse map during a reorg while fetching headers within the chain's checkpointed
+	// region.
+	hdrTipSlice []int32
```

Comment:
I guess my reasoning behind this was that in the `processBlKHeaderInCheckPtRegionInOrder` function, I am literally mapping the responses gotten at each header tip at that point to header tip itself but I guess it can be refactored.

Reply:
I'm not suggesting refactoring in this particular case. Though I think the PR overall would benefit from being broken up a bit more. In this case I'm more commenting on the fact that the name of this field doesn't describe the field's purpose very well, making it hard to figure out what it's being used for. Can you define for me precisely what a "header tip" is in the way you are using it?

Reply:
https://github.com/Chinwendu20/neutrino/blob/64b278771ff75da0b30136af7d4ab0ace06d897e/blockmanager.go#L2166-L2179

Because at the point of writing into the map, I am literally mapping the response gotten at each header tip i.e. chain tip to the chain tip (header tip) itself but if that is not clear, I could refactor (rename) the slice.

Reply:
I think the docs added to the comment could help in understanding what the slice is about.

Reply:
rename this to `regionTips` since it represents the header tip for each region.